ygen/protogen.go

```diff
@@ -819,19 +819,8 @@ func writeProtoEnums(enums map[string]*yangEnum, annotateEnumNames bool) ([]stri
 			}
 			p.Values = ge.Values
 
-			// If the supplied enum entry has the valuePrefix annotation then use it to
-			// calculate the enum value names.
+			// Capitalize name per proto style.
 			p.ValuePrefix = strings.ToUpper(enum.name)
-			if e, ok := enum.entry.Annotation["valuePrefix"]; ok {
```

Comment:
Can we make this be behind a flag? (It looks like today this will be independent of whether `-typedef_enum_with_defmod` is set or not) 

Given that we want users to be able to be backwards compatible here if they need to regenerate their protos and don't want this change - then I suspect we maybe have to have another flag -- if so, we should also fix the suffixing issue before exposing the change to the protogenerator binary.

Reply:
A flag is added, where the "Enum" suffix is now added for simple union enums, and the redundant union enums have also been removed from the generated code (please see first section of the newly updated PR description)

Reply:
Great, thanks!